### PR TITLE
feature: Add MySQL as migration and validation source for ScyllaDB

### PR DESCRIPTION
**Pull request by dkropachev**
## Summary
Add MySQL as a migration and validation source for ScyllaDB, with comprehensive security hardening, config validation, and hash-based validation support.

## Context
The scylla-migrator previously supported Cassandra, DynamoDB, and Parquet as migration sources. This PR adds MySQL as a new source type, enabling users to migrate data from MySQL databases to ScyllaDB via Spark JDBC. A post-migration validator is also included to verify data integrity using either full row comparison or content-hash-based comparison for large tables.

### Core Migration Support
- New `SourceSettings.MySQL` configuration type with full YAML parsing, defaults, and extensive parse-time validation (~20 checks including host/database/table format, port range, credential non-emptiness, partition consistency, SQL injection prevention)
- JDBC reader (`readers/MySQL.scala`) with hardened URL construction, cursor-based fetching, and partitioned read support
- `MySQLSchemaLogger` for diagnostic logging of MySQL-to-CQL type mappings
- Wiring in `Migrator.scala` and `Validator.scala` for the `(MySQL, Scylla)` source/target combination

### Validation
- `MySQLToScyllaValidator` with full-outer join approach, column rename support, case-insensitive field resolution, and content-hash mode (`MD5`-based, computed in Spark on both sides to avoid cross-system hash divergence)
- `hashColumns` config option in `Validation` to reduce shuffle volume for wide tables
- Shared `RowComparisonFailure.areDifferent` with tolerance support for timestamps, floats, BigDecimals, NaN/Infinity, and byte arrays

### Security Hardening
- 21 dangerous JDBC properties blocked (`allowLoadLocalInFile`, `autoDeserialize`, `queryInterceptors`, etc.)
- Dangerous SQL keyword detection in WHERE clauses upgraded to hard reject with expanded keyword list (DROP, DELETE, TRUNCATE, ALTER, CREATE, EXEC, EXECUTE, UNION, INTO, OUTFILE, DUMPFILE, LOAD_FILE, BENCHMARK, SLEEP, GRANT, REVOKE)
- SQL comment stripping before keyword check to prevent bypass via `UNI/**/ON`
- Control character blocking in WHERE clauses
- Password redaction in savepoint YAML output with clear error when loading redacted savepoints
- Connection property redaction for sensitive keys (`password|secret|token|credential`)
- TLS keystore URL scheme warnings
- `spark.redaction.regex` configuration warnings

### Code Review Fixes (second commit)
- Narrowed `connectionProperties` redaction regex from `password|secret|key` to `password|secret|token|credential` to avoid false positives on keys like `sslKeyManagerFactory`
- Replaced `df.rdd.getNumPartitions` with `df.queryExecution.executedPlan.execute().getNumPartitions` to avoid RDD conversion overhead
- Added `fetchSize > 10000` warning for memory-sensitive workloads
- Removed trivial `valuesAreDifferent` wrapper, calling `RowComparisonFailure.areDifferent` directly
- Fixed incorrect test comment about `timestampMsTolerance = 0` code path

## Testing
Unit tests (no external services required):
```bash
make test-unit
```

All 283 unit tests pass, including:
- `MySQLSourceSettingsParserTest` (35 tests) - config parsing, validation failures, edge cases
- `MySQLReaderTest` - `escapeIdentifier`, `buildJdbcUrl`, host validation
- `MySQLToScyllaValidatorTest` - value comparison, prefix columns, content hash computation
- `MySQLConfigSerializationTest` (3 tests) - YAML render/redaction verification
- `ScyllaSparkConnectionOptionsTest` - target connection options

Linting:
```bash
make lint
```

### CODE REVIEW

**Comment by tarzanek** [comment]
so this should be superior to https://github.com/scylladb/scylla-migrator/pull/343, let's check it

File: build.sbt [hunk 1]
```diff
@@ -61,6 +61,7 @@ lazy val migrator = (project in file("migrator"))
       "com.github.jnr" % "jnr-posix" % "3.1.19", // Needed by the Spark ScyllaDB connector
       "com.scylladb.alternator" % "emr-dynamodb-hadoop"  % "5.8.0",
       "com.scylladb.alternator" % "load-balancing"       % "2.0.3",
+      "com.mysql"              % "mysql-connector-j"     % "9.1.0",
```
Comment (tarzanek) [inline build.sbt]: 9.7.0 is latest, why 9.1.0 is used?

Comment (tarzanek) [reply build.sbt]: 8.4.0 is latest LTS, 9.7.0 is latest new one

support wise 8.x supports 
5.7+ (Earlier 8.x), 8.0/8.4 

while 9.x supports  8.0+ (8.0, 8.4 LTS, 9.x)
